src/generated/info/mod.rs

```diff
@@ -34,71 +33,32 @@ impl From<&info::Version> for Version {
-            info::InfoResult_Result::UNKNOWN => Result::Unknown,
-            info::InfoResult_Result::SUCCESS => Result::Success,
-            info::InfoResult_Result::INFORMATION_NOT_RECEIVED_YET => {
-                Result::InformationNotReceivedYet
-            }
-        }
-    }
-}
-
-impl Into<info::InfoResult_Result> for Result {
-    fn into(self) -> info::InfoResult_Result {
-        match self {
-            Result::Unknown => info::InfoResult_Result::UNKNOWN,
-            Result::Success => info::InfoResult_Result::SUCCESS,
-            Result::InformationNotReceivedYet => {
-                info::InfoResult_Result::INFORMATION_NOT_RECEIVED_YET
-            }
-        }
-    }
-}
-
-#[derive(PartialEq, Clone, Debug)]
-pub struct InfoResult {
-    result: Result,
-    result_str: String,
-}
+pub type GetVersionResult = super::super::RequestResult<Version, InfoError>;
 
-impl From<&info::InfoResult> for InfoResult {
-    fn from(grpc_info_result: &info::InfoResult) -> Self {
-        Self {
-            result: Result::from(grpc_info_result.get_result()),
-            result_str: String::from(grpc_info_result.get_result_str()),
-        }
-    }
-}
-
-impl Into<info::InfoResult> for InfoResult {
-    fn into(self) -> info::InfoResult {
-        let mut rpc_info_result = info::InfoResult::default();
-        rpc_info_result.set_result(self.result.into());
-        rpc_info_result.set_result_str(self.result_str);
-        rpc_info_result
-    }
-}
-
-#[derive(PartialEq, Clone, Debug)]
-pub struct GetVersionResponse {
-    info_result: InfoResult,
-    version: Version,
-}
-
-impl From<info::GetVersionResponse> for GetVersionResponse {
-    fn from(rpc_get_version_response: info::GetVersionResponse) -> Self {
-        Self {
-            info_result: InfoResult::from(rpc_get_version_response.get_info_result()),
-            version: Version::from(rpc_get_version_response.get_version()),
+impl FromRpcResult<info::GetVersionResponse> for GetVersionResult {
+    fn from_rpc_result(
+        rpc_get_version_response: ::grpcio::Result<info::GetVersionResponse>,
+    ) -> Self {
+        match rpc_get_version_response {
+            Ok(verison_response) => match verison_response.get_info_result().get_result() {
+                info::InfoResult_Result::UNKNOWN => Err(super::super::RequestError::MavErr(
+                    InfoError::Unknown(verison_response.get_info_result().get_result_str().into()),
+                )),
+                info::InfoResult_Result::INFORMATION_NOT_RECEIVED_YET => Err(
+                    super::super::RequestError::MavErr(InfoError::InformationNotReceivedYet(
+                        verison_response.get_info_result().get_result_str().into(),
+                    )),
+                ),
+                info::InfoResult_Result::SUCCESS => {
+                    Ok(Version::from(verison_response.get_version()))
+                }
+            },
+            Err(e) => Err(super::super::RequestError::RpcErr(e)),
```

Comment:
So there is new draft to generate template to [this type ](https://github.com/mavlink/MAVSDK-Rust/pull/4/files#diff-b4aea3e418ccdb71239b96952d9cddb6R14-R16)